src/list.c

```diff
@@ -88,4 +85,12 @@ void list_insert(struct list *list, void *item, struct node *after)
 	new->next = *x;
 	*x = new;
 	new->prev = after;
+	return true;
 }
+
+
+void list_append_ctx(struct list *list, void *item, struct led_ctx *ctx)
+{
+	if (!list_insert(list, item, list->tail))
+		ctx->deferred_error = LED_STATUS_OUT_OF_MEMORY;
```

Comment:
The value is returned as the final value in `led_scan` and simplifies the code so that we don't need to bubble up out of memory errors across the scan.  This is done via `list_append_ctx`.

```C
led_status_t led_scan(struct led_ctx *ctx)
{
	if(!ctx)
		return LED_STATUS_NULL_POINTER;

	ctx->deferred_error = LED_STATUS_SUCCESS;

	sysfs_reset(ctx);
	sysfs_scan(ctx);

	return ctx->deferred_error;
}
```